File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/sample/search/ListableSampleTypeSearchCriteria.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractSearchCriteria;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.sample.search.ListableSampleTypeSearchCriteria")
public class ListableSampleTypeSearchCriteria extends AbstractSearchCriteria
{
    private static final long serialVersionUID = 1L;

    private boolean listable;

    public ListableSampleTypeSearchCriteria()
    {

    }

    public void thatEquals(boolean value)
    {
        setListable(value);
    }

    public void setListable(boolean value)
    {
        this.listable = value;
    }

    public boolean isListable()
    {
        return listable;
    }

    @Override
    public String toString()
    {
        return "listable: " + isListable();
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/sample/search/ListableSampleTypeSearchCriteria.js
define([ "stjs", "as/dto/common/search/AbstractSearchCriteria" ], function(stjs, AbstractSearchCriteria) {
	var ListableSampleTypeSearchCriteria = function() {
		AbstractSearchCriteria.call(this);
	};
	stjs.extend(ListableSampleTypeSearchCriteria, AbstractSearchCriteria, [ AbstractSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.sample.search.ListableSampleTypeSearchCriteria';
		constructor.serialVersionUID = 1;
		
		prototype.listable;
	    
		prototype.thatEquals = function(value) {
	        this.setListable(value);
	    }

		prototype.setListable = function(value) {
	        this.listable = value;
	    }

		prototype.isListable = function()
	    {
	        return this.listable;
	    }

	}, {});
	return ListableSampleTypeSearchCriteria;
})
